glibc/plan.sh

```diff
@@ -322,6 +322,29 @@ EOF
   popd > /dev/null
 }
 
+do_strip() {
+  build_line "Stripping unneeded symbols from binaries and libraries"
+  find "$pkg_prefix" -type f -perm -u+w -print0 2> /dev/null \
+    | while read -rd '' f; do
+      case "$(basename "$f")" in
+        "ld-${pkg_version}.so"|\
+        "libc-${pkg_version}.so"|\
+        "libpthread-${pkg_version}.so"|\
+        libpthread_db-1.0.so)
+          build_line "Skipping strip for $f"
+          continue
+          ;;
+      esac
+
+      case "$(file -bi "$f")" in
+        *application/x-executable*) strip --strip-all "$f";;
+        *application/x-sharedlib*) strip --strip-unneeded "$f";;
+        *application/x-archive*) strip --strip-debug "$f";;
+        *) continue;;
```

Comment:
Given the issues we encountered in habitat-sh/habitat#6184, I think we should add `*application/x-pie-executable*) strip --strip-all "$f";;` here so that we are consistent with where plan-build will land as we progress with this refresh.

Reply:
Got it. Will work on that

Reply:
@smacfarlane i think you meant https://github.com/habitat-sh/habitat/pull/6189 -- however it looks like it is `--strip-unnedded` instead of what you recommend?

Reply:
:man_facepalming: Yes, habitat-sh/habitat#6189 is the correct issue.  

The original intent of the `do_default_strip` was for `--strip-all` on all executables. Due to the bug in file they were being considered as sharedlib and were getting `--strip-unneeded`.  When we updated `do_default_strip` we elected to stay consistent with existing behavior and fix it to match original intent around the time of a base plans refresh.

That's a long way of saying `--strip-all` is probably what we want, but may need to adjust as we iterate on the refresh.

Reply:
I see. Got it. `--strip-all` is in place.